### ISSUE TITLE
parfor fails running FT (multiple versions issue?)

### ISSUE THREAD

**Issue by NirOfir** [fieldtrip/fieldtrip#2206]
**Describe the bug**
I seem to be having the same ft_version() issue described earlier in #1249. I'm not entirely sure how to reliably reproduce it. The first few "iterations" in the parfor run OK, but then I get this error:

```
Error using ft_version (line 161)
Insufficient number of outputs from right hand side of equal sign to satisfy assignment.

Error in ft_notification

Error in ft_warning (line 63)
  ft_notification(varargin{:});

Error in ft_defaults>checkMultipleToolbox (line 405)
  ft_warning('Multiple versions of %s on your path will confuse FieldTrip', toolbox);

Error in ft_defaults (line 201)
  checkMultipleToolbox('FieldTrip',           'ft_defaults.m');
```

I added a line to ft_defaults() right after 'if length(list)>1' in line 403, to see what the toolbox list variable holds. This is the output I got:

```
list =

  2x1 cell array

    {'/tmp/tp2bb36de4_2a22_4a2a_b6f0_9ef2996c838crp16727/a/tpc1c30f10_f249_4420_8f4e_5967168bb410/ft_defaults.m'}
    {'/mnt/raid0/Ongoing Timing/fieldtrip-20220628/ft_defaults.m' 
```

In the function called by the parfor loop, I call these FT functions (in the listed order):
ft_rejectcomponent()
ft_selectdata()
ft_channelrepair()
ft_scalpcurrentdensity()
ft_freqanalysis()
ft_math()

Reading through the command line output, it seems like the bug is related to ft_scalpcurrentdensity(), becuase I don't get any ft_freqanalysis() or ft_math() feedback. More specifically, I only get the following line once:

> output surface laplacian is in V/m^2

After adding debug print lines here and there in the suspect FT functions, it looks like the preamble calls in ft_freqanalysis is where things go wrong:
https://github.com/fieldtrip/fieldtrip/blob/b717eba0408055ef5412d1a8f5c5c2fcdc4d620f/ft_freqanalysis.m#L230-L235

Any ideas on what might be going on here, and how to move forward with this?

* If I just start FT fresh without running any FT functions and call ft_version() a few times inside a parfor loop it works fine. 
* I used parfor pretty extensively with an older version of FT (20210623) without issues, but using the older version with the current code doesn't help. 

**Environment and versions (please complete the following information):**
 - Operating System: CentOS 8
 - MATLAB version R2018b
 - FieldTrip version 20220628

Thanks!

**Comment by robertoostenveld** [comment]
this is pertinent to https://github.com/fieldtrip/fieldtrip/projects/5 and relevant for the work that @AljenU is doing, among others https://github.com/fieldtrip/website/pull/630. Thanks for tracking it down so far already!

You have reference to a temporary file `/tmp/tp2b...68bb410/ft_defaults.m`, which suggests that the code gets physically distributed (since copied). Are you running parfor on a local cluster (i.e., multi-CPU machine) or on a  MATLAB parallel server?

**Comment by NirOfir** [comment]
Local cluster, but I also have some happy news: Since I couldn't get parfor to work, I switched back to a simple for, and found that I had misnamed a variable in the call to ft_freqanalysis(). It seems like matlab is returning this cryptic error message instead of the expected 'Unrecognized function or variable' error message. I don't know which other errors are also messed up by matlab's parfor, but I did notice before that the errors I get from parfor usually aren't very informative. It now turns out that they can also be misleading :)

So, a post in the FAQ or a couple of lines in the relevant doc page, is probably the solution :)
